privacyidea/api/lib/prepolicy.py

```diff
@@ -506,6 +506,31 @@ def init_token_defaults(request=None, action=None):
     return True
 
 
+def init_registrationcode_length_contents(request=None, action=None):
+    """
+    This policy function is to be used as a decorator in the API token init function.
+
+    If there is a valid policy set the action values of REGISTRATIONCODE_LENGTH
+    and REGISTRATIONCODE_CONTENTS are added to request.all_data as
+
+    { 'registrationcode_length': '10', 'registrationcode_contents': 'cn' }
+    """
+    params = request.all_data
+    tokentype = params.get("type")
+    if tokentype == "registration":
+        user_object = get_user_from_param(params)
```

Comment:
You should set a default length and contents here:

~~~~
request.all_data["registration.length"] = 24
request.all_data["registration.length"] = please_check_what_the_default_is
~~~~

Otherwise a user or administrator will be able to manuall set a short registration code length if no policy is given. This is what I ment in my comment https://github.com/privacyidea/privacyidea/issues/2284#issuecomment-725384088

Reply:
I set the defaults in registrationtoken.py and import them where needed. 
I think the default contents were "cns". But since the length is already 24, I now set it to "cn" only.